addons/web_editor/static/src/components/media_dialog/image_selector.js

```diff
@@ -162,7 +162,35 @@ export class ImageSelector extends FileSelector {
 
     validateUrl(...args) {
         const { isValidUrl, path } = super.validateUrl(...args);
-        const isValidFileFormat = IMAGE_EXTENSIONS.some(format => path.endsWith(format));
+        // Validate that the URL is an image
+        const isValidImage = async () => {
+            return fetch(path, { method: 'HEAD' })
+            .then(response => {
+                const contentType = response.headers.get('content-type');
+                return contentType && IMAGE_MIMETYPES.includes(contentType);
+            })
+            .catch(() => {
+                /**
+                 * A fallback that handles CORS, or Cross-Origin Resource
+                 * Sharing, is a mechanism that uses additional HTTP headers
+                 * to tell browsers to give a web application running at one
+                 * origin, access to selected resources from a different origin.
+                 * In the context of this code, if the server hosting the image
+                 * doesn't allow cross-origin requests, the fetch request will
+                 * fail due to the browser's same-origin policy, even if the
+                 * image URL is valid and the image exists.In that case we
+                 * attempt to check whether the image exists by creating an
+                 * Image object and checking its onload and onerror events.
+                */
+                return new Promise((resolve) => {
+                    const img = new Image();
+                    img.src = path;
+                    img.onload = () => resolve(true);
+                    img.onerror = () => resolve(false);
+                });
+            });
+        }
+        const isValidFileFormat = isValidUrl && isValidImage().then(isValidImageResult => isValidImageResult);
```

Comment:
@shsa-odoo This is better than the previous version but the promise chaining is still wrong and that is why it doesn't work in testing.

```suggestion
    async validateUrl(...args) {
        const { isValidUrl, path } = super.validateUrl(...args);
        const isValidFileFormat = isValidUrl && await new Promise(resolve => {
            // Validate that the URL is an image.
            fetch(path, { method: 'HEAD' })
            .then(response => {
                const contentType = response.headers.get('content-type');
                resolve(contentType && IMAGE_MIMETYPES.includes(contentType));
            })
            .catch(() => {
                /**
                 * If the server hosting the image doesn't allow cross-origin
                 * requests, the fetch request will fail due to the browser's
                 * same-origin policy, even if the image URL is valid and the
                 * image exists. In that case we trade-off the check by creating
                 * an Image object and checking its onload and onerror events.
                */
                const img = new Image();
                img.src = path;
                img.onload = () => resolve(true);
                img.onerror = () => resolve(false);
            });
        });
    }
```
Then in `file_selector.js` you'll need to make `validateUrl` async as well, then change `onUrlInput` like so:
```js
async onUrlInput(ev) {
        const { isValidUrl, isValidFileFormat } = await this.props.validateUrl(ev.target.value);
```
And then I think it should work.

In addition, I don't think the other commit with the changes in Python is required.

There are 12 cases to test for this PR, I marked the behaviors that are new to this PR with "New!":
```
+---------------------------------------+------------+-------------+---------+------------+
| File Type                             | Extension? | CORS Error? | Upload? | Notes      |
+---------------------------------------+------------+-------------+---------+------------+
| Supported image                       | Yes        | No          | Yes     |            |
+---------------------------------------+------------+-------------+---------+------------+
| Supported image                       | No         | No          | Yes     | New!       |
+---------------------------------------+------------+-------------+---------+------------+
| Unsupported image like JPEG XL (.jlx) | Yes        | No          | No      |            |
+---------------------------------------+------------+-------------+---------+------------+
| Unsupported image like JPEG XL (.jlx) | No         | No          | No      | New!       |
+---------------------------------------+------------+-------------+---------+------------+
| Not image                             | Yes        | No          | No      |            |
+---------------------------------------+------------+-------------+---------+------------+
| Not image                             | No         | No          | No      | New!       |
+---------------------------------------+------------+-------------+---------+------------+
| Supported image                       | Yes        | Yes         | Yes     |            |
+---------------------------------------+------------+-------------+---------+------------+
| Supported image                       | No         | Yes         | Yes     | New!       |
+---------------------------------------+------------+-------------+---------+------------+
| Unsupported image like JPEG XL (.jlx) | Yes        | Yes         | No      |            |
+---------------------------------------+------------+-------------+---------+------------+
| Unsupported image like JPEG XL (.jlx) | No         | Yes         | Yes     | Trade-off! |
+---------------------------------------+------------+-------------+---------+------------+
| Not image                             | Yes        | Yes         | No      |            |
+---------------------------------------+------------+-------------+---------+------------+
| Not image                             | No         | Yes         | No      | New!       |
+---------------------------------------+------------+-------------+---------+------------+
```

Reply:
@dmo-odoo Sorry for the delayed response, Yes those changes will solve the problem, but it will also add a wierd scenario [video](https://www.awesomescreenshot.com/video/26343911?key=b488daf89a829c7c65a2cdff20523236), initially the url entered will be shown invalid and when the promise resolves it will show it valid, which i think we will need to fix as well.

> In addition, I don't think the other commit with the changes in Python is required.

IMO adding mimetype to all those image url (if valid image) that doesnot ends with valid image extensions because they contains unsupported mimetype in them, when we test it for our case url ([https://www.odoo.com/web/image/6049005](https://github.com/web/image/6049005))
see image :

![image](https://github.com/odoo/odoo/assets/103405229/f3d9d43c-be02-4fc8-8cfb-4d3419c1664a)


 as our current code [here](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/models/ir_attachment.py#L41) adds false to attachment src.

Reply:
I understand the use case yes. I'll ask the security team to validate when we're ready.

Reply:
About your scenario, I don't know where this invalid sign comes from, but I think there should be no validation sign at all until the promise resolves.

Reply:
@dmo-odoo i looked into the issue and found that when we make `onUrlInput` async the call on input [here](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/components/media_dialog/file_selector.xml#L20C292-L20C302), the state variables we recieve from this method initally identify that url as invalid as these variables remains undefined, and when promise resolves and the value of this variable is updated and value which changes [o_we_url_error](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/components/media_dialog/file_selector.xml#L26C74-L26C88) to [o_we_url_success](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/components/media_dialog/file_selector.xml#L25C101-L25C117) what is you view on this ?

Reply:
> @dmo-odoo i looked into the issue and found that when we make `onUrlInput` async the call on input [here](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/components/media_dialog/file_selector.xml#L20C292-L20C302), the state variables we recieve from this method initally identify that url as invalid as these variables remains false, and when promise resolves and the value of this variable is updated and value which changes [o_we_url_error](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/components/media_dialog/file_selector.xml#L26C74-L26C88) to [o_we_url_success](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/components/media_dialog/file_selector.xml#L25C101-L25C117) what is you view on this ?

@dmo-odoo ,  since it was the requirement of our code to make `onUrlInput` async as our data regarding `isValidFileFormat` is coming from fetching the url,i have added another check in xml that will showcase the loading effect while our url is getting validated. i have made the changes and it is way better that initally showing  :x:  mark while we validate the url and then show the :heavy_check_mark:  if its validates to true.
 What is your view on this?  Thanks!
 cc @deso-odoo